### PR TITLE
Add satisfaction survey link to help menu

### LINKED ISSUE #3521: Add Link to UX survey into Dashboards
**Is your feature request related to a problem? Please describe.**

**What:** We are proposing that we add a link to the open source survey into the Dashboards open source experience. 
**Why:**  We need user input on the dashboard creation and consumption experience. Users intercepted and providing feedback on features elsewhere (such as [OpenSearch.org](http://opensearch.org/) etc.) lack the specificity required to improve the dashboarding experience.

image

image

### PR DESCRIPTION
**Pull request by abbyhu2000**
Add a link to the survey under the help menu on the top navigation bar. Also change the helper menu UI according to issue #3521 

The link can be configurable in config/opensearch_dashboards.yml
`#opensearchDashboards.survey.url: "https://survey.opensearch.org"`

1. The default is to show the survey `Give feedback` in the helper menu
Screen Shot 2023-03-30 at 4 31 10 PM

2. If admin wishes to disable the survey, they can edit the value in `opensearch_dashboards.yml` and set: 
`opensearchDashboards.survey.url: ""`
Screen Shot 2023-03-30 at 4 31 41 PM

3. If admin wishes to configure the survey, they can change the link in `opensearch_dashboards.yml` and set:
`opensearchDashboards.survey.url: "/newSurveyLink"`

### Issues Resolved
resolves #3521 
 
### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by ashwin-pc — DISMISSED** [review]

**Review by kavilla — DISMISSED** [review]
I think I would want the ability to at minimum disable this feature.

The next level would be able to configure the link utilized.

**Comment by abbyhu2000** [comment]
> I think I would want the ability to at minimum disable this feature.
> 
> The next level would be able to configure the link utilized.

Added a config `#opensearchDashboards.isSurveyAllowed: true` in the yml file to enable/disable the survey link. @kavilla

File: config/opensearch_dashboards.yml [hunk 1]
```diff
@@ -237,3 +237,5 @@
 #data_source.encryption.wrappingKeyName: 'changeme'
 #data_source.encryption.wrappingKeyNamespace: 'changeme'
 #data_source.encryption.wrappingKey: [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
+
+#opensearchDashboards.isSurveyAllowed: true
```
Comment (kavilla) [inline config/opensearch_dashboards.yml:241]: to be honest i think breaking this down into more objects is more scalable and matches the structure of the project.

for example:
```
opensearchDashboards.survey.enabled: true
```

Comment (abbyhu2000) [reply config/opensearch_dashboards.yml]: comment addressed here https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3521#issuecomment-1487788462

File: src/legacy/server/config/schema.js [hunk 2]
```diff
@@ -247,6 +247,9 @@ export default () =>
         applicationTitle: Joi.any().default(''),
         useExpandedHeader: Joi.boolean().default(true),
       }),
+      survey: Joi.object({
+        url: Joi.any().default('/'),
```
Comment (kavilla) [inline src/legacy/server/config/schema.js:251]: why not: `https://amazonmr.au1.qualtrics.com/jfe/form/SV_1BxJNrtCo4LbweW`

Same as the default value above.

File: src/core/public/chrome/ui/header/header_help_menu.tsx [hunk 3]
```diff
@@ -122,6 +122,7 @@ interface Props {
   opensearchDashboardsVersion: string;
   useDefaultContent?: boolean;
   opensearchDashboardsDocLink: string;
+  survey?: string;
```
Comment (kavilla) [inline src/core/public/chrome/ui/header/header_help_menu.tsx:125]: nit:

line `124` says `opensearchDashboardsDocLink` then for consistency I think it should be `surveyLink` if this value is the link.

File: src/legacy/server/config/schema.js (hunk 2 above)
Comment (ashwin-pc) [reply src/legacy/server/config/schema.js:251]: I dont think we should add the amazon survey link in the code itself

File: config/opensearch_dashboards.yml [hunk 4]
```diff
@@ -237,3 +237,5 @@
 #data_source.encryption.wrappingKeyName: 'changeme'
 #data_source.encryption.wrappingKeyNamespace: 'changeme'
 #data_source.encryption.wrappingKey: [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
+
+#opensearchDashboards.survey.url: "https://amazonmr.au1.qualtrics.com/jfe/form/SV_1BxJNrtCo4LbweW"
```
Comment (joshuarrrr) [inline config/opensearch_dashboards.yml:241]:
```suggestion

# Set the value of this setting to false to hide the help menu link to the OpenSearch Dashboards user survey
# opensearchDashboards.survey.url: "https://amazonmr.au1.qualtrics.com/jfe/form/SV_1BxJNrtCo4LbweW"
```

**Review by joshuarrrr — COMMENTED** [review]
Some minor changes requested

**Comment by seanneumann** [comment]
Did we investigate setting the url to something like "survey.opensearch.org" so the survey endpoint can be updated via DNS?

**Comment by AMoo-Miki** [comment]
> Did we investigate setting the url to something like "survey.opensearch.org" so the survey endpoint can be updated via DNS?

This would be exceptionally simple to do. Though, I would prefer some specificity in the URL so we can use the domain for other stuff too; so while the domain will be `survey.opensearch.org`, at runtime we would append `osd-version=2.7.0` which will tell us that (1) it came from OSD and (2) it was OSD 2.7.0.

**Comment by seanneumann** [comment]
I like it Miki. let's do it. 

Any other params to append?

File: src/core/public/chrome/ui/header/header_help_menu.tsx [hunk 5]
```diff
@@ -328,7 +332,7 @@ class HeaderHelpMenuUI extends Component<Props, State> {
         })}
         onClick={this.onMenuButtonClick}
       >
-        <EuiIcon type="help" size="m" />
+        <EuiIcon type="questionInCircle" size="l" />
```
[the remark this answers is not in the document]
Comment (abbyhu2000) [reply src/core/public/chrome/ui/header/header_help_menu.tsx:335]: Need confirmation from the UX team, @KrooshalUX for his help menu icon, should we make it size 'm' or 'l'? Size 'm' looks kinda small to me

File: src/core/public/chrome/ui/header/header_help_menu.tsx [hunk 6]
```diff
@@ -215,23 +229,13 @@ class HeaderHelpMenuUI extends Component<Props, State> {
         <EuiButtonEmpty href={helpSupportUrl} target="_blank" size="xs" flush="left">
           <FormattedMessage
             id="core.ui.chrome.headerGlobalNav.helpMenuAskElasticTitle"
```
[the remark this answers is not in the document]
Comment (abbyhu2000) [reply src/core/public/chrome/ui/header/header_help_menu.tsx:231]: Good call~

File: src/core/public/chrome/ui/header/header.test.tsx [hunk 7]
```diff
@@ -75,6 +75,7 @@ function mockProps() {
       mark: { defaultUrl: '/' },
       applicationTitle: 'OpenSearch Dashboards',
     },
+    survey: '/',
```
[the remark this answers is not in the document]
Comment (abbyhu2000) [reply src/core/public/chrome/ui/header/header.test.tsx:78]: There is no unit tests that are covering the helper menu itself. Added a test file `header_help_menu.test.tsx` and added two unit tests for displaying and hiding the survey link. @joshuarrrr

File: src/core/public/chrome/ui/header/header_help_menu.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (abbyhu2000) [reply src/core/public/chrome/ui/header/header_help_menu.tsx:335]: @kgcreative @joshuarrrr  For reference, the first image is size 'l' and the second image is size 'm'. 
Screen Shot 2023-04-05 at 4 04 00 PM
Screen Shot 2023-04-05 at 4 04 36 PM

**Review by kavilla — DISMISSED** [review]

File: src/core/public/chrome/ui/header/header_help_menu.tsx [hunk 8]
```diff
@@ -198,15 +199,28 @@ class HeaderHelpMenuUI extends Component<Props, State> {
...
       useDefaultContent,
       opensearchDashboardsDocLink,
+      survey,
     } = this.props;
     const { helpExtension, helpSupportUrl } = this.state;
+    const opensearchSurvey = survey ? (
+      <div>
+        <EuiButtonEmpty href={survey} target="_blank" size="xs" flush="left">
+          <FormattedMessage
+            id="core.ui.chrome.headerGlobalNav.helpMenuSatisfactionSurveyTitle"
```
[the remark this answers is not in the document]
Comment (abbyhu2000) [reply src/core/public/chrome/ui/header/header_help_menu.tsx]: changed

**Comment by abbyhu2000** [comment]
Currently the cypress tests for multiple datasource are failing, added a line in `cypress_workflow.yml` to skip multiple data source cypress tests so the cypress tests action can pass. Created an issue to solve the cypress test failure.https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3843
Screenshot 2023-04-13 at 1 17 41 PM

File: .github/workflows/cypress_workflow.yml [hunk 9]
```diff
@@ -12,7 +12,7 @@ env:
   START_CMD: 'node ../scripts/opensearch_dashboards --dev --no-base-path --no-watch'
   OPENSEARCH_SNAPSHOT_CMD: 'node ../scripts/opensearch snapshot'
   SPEC: 'cypress/integration/core-opensearch-dashboards/opensearch-dashboards/**/*.js,'
-  CYPRESS_ENV: 'env CYPRESS_VISBUILDER_ENABLED=true '
+  CYPRESS_ENV: 'env CYPRESS_VISBUILDER_ENABLED=true,DATASOURCE_MANAGEMENT_ENABLED=false'
```
Comment (abbyhu2000) [inline .github/workflows/cypress_workflow.yml:15]: This can be removed once the MD cypress tests are fixed. #3843

**Comment by abbyhu2000** [comment]
Also Link checker fail because it references Roubenmeschian and Roubenmeschian server is down. Will delete Roubenmeschian reference in the code. (it exists in two comments in discover plugin)

Screenshot 2023-04-13 at 1 36 06 PM
Screenshot 2023-04-13 at 1 37 46 PM

**Review by kavilla — DISMISSED** [review]

File: src/plugins/discover/public/application/angular/doc_table/create_doc_table_react.tsx [hunk 10]
```diff
@@ -87,8 +87,6 @@ export function convertDirectiveToRenderFn(
 
     return () => {
       if (!rejected) {
-        // for cleanup
-        // http://roubenmeschian.com/rubo/?p=51
```
Comment (AMoo-Miki) [inline src/plugins/discover/public/application/angular/doc_table/create_doc_table_react.tsx:91]: This change is now in 3 PRs.

File: src/plugins/discover/public/application/angular/doc_table/create_doc_table_react.tsx [hunk 11]
```diff
@@ -87,8 +87,6 @@ export function convertDirectiveToRenderFn(
 
     return () => {
       if (!rejected) {
-        // for cleanup
```
Comment (AMoo-Miki) [inline src/plugins/discover/public/application/angular/doc_table/create_doc_table_react.tsx:90]: This change is now in 3 PRs. Let's rebase this PR after #3847 is merged.

**Review by AMoo-Miki — CHANGES_REQUESTED** [review]

File: src/plugins/discover/public/application/angular/doc_table/create_doc_table_react.tsx (hunk 11 above)
Comment (abbyhu2000) [reply src/plugins/discover/public/application/angular/doc_table/create_doc_table_react.tsx]: done

**Review by ananzh — APPROVED** [review]

**Review by AMoo-Miki — APPROVED** [review]

**Pull request merged**